fix: Filter empty strings out of list options

ValeriaVG opened the pull request:
Filters out empty strings from `split` list env variables, allowing `TS_NODE_IGNORE=  <...>` to be treated as an empty array.

Closes #1301

cspotcode commented:
Thanks, where does this `split` function get used?  Is it *only* used to split environment variables?  Just want to be sure we don't break anything else.

Any chance you can tweak the JSDoc comment a bit to describe what it does, or at least explain that it's meant for env var parsing?  Looks like it does some whitespace removal too.  It might be nice to further clarify its intended usage, now that it's doing even more than a string split.

ValeriaVG commented:
@cspotcode Yes, it's imported only in `src/index.ts` and used only here:
```ts
 ignore: split(env.TS_NODE_IGNORE),
 // ...
 ignoreDiagnostics: split(env.TS_NODE_IGNORE_DIAGNOSTICS),
```

cspotcode reviewed (approved):
Approved.  Pretty sure the test failures are some unrelated test flakiness that I'll address somewhere else.

cspotcode commented:
@ValeriaVG thanks again!
